### PR TITLE
Add Jest Unit Testing compatibility

### PR DESCRIPTION
**Pull request by shucon**
### Description

Fixes: #16405

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -68,6 +68,14 @@ jobs:
       after_success: skip
       script: composer psalm
 
+    - stage: "JavaScript Unit Test"
```
Comment (williamdes) [inline .travis.yml:71]:
```suggestion
    - stage: "Other tests"
```

Please use a stage defined in the stages block at top of this file

File: js/src/ajax.js [hunk 2]
```diff
@@ -971,3 +971,7 @@ $(document).on('ajaxError', function (event, request) {
         AJAX.xhr = null;
     }
 });
+
+if (typeof exports !== 'undefined') {
+    module.exports = AJAX;
+}
```
Comment (mauriciofauth) [inline js/src/ajax.js:977]: Is this really necessary?

Comment (shucon) [reply js/src/ajax.js:977]: Yes, `module.exports` is required for the tests to work AFAIK. Jest was throwing `AJAX is not defined` errors without it.

Comment (mauriciofauth) [reply js/src/ajax.js:977]: `error  'module' is not defined`
https://travis-ci.org/github/phpmyadmin/phpmyadmin/jobs/735793795#L444

Comment (shucon) [reply js/src/ajax.js]: Is it fine now?

File: .travis.yml [hunk 3]
```diff
@@ -68,6 +68,14 @@ jobs:
       after_success: skip
       script: composer psalm
 
+    - stage: "Other tests"
+      name: "Run Jest Unit tests"
+      before_install: phpenv config-rm xdebug.ini
```
Comment (williamdes) [inline .travis.yml:73]:
```suggestion
      before_install: skip
```

**Review by williamdes — APPROVED** [review]
Good starting point, one thing bugs me and it is the imports blocks added to the source files
I am sure you could do someway around this

**Comment by shucon** [comment]
> Good starting point, one thing bugs me and it is the imports blocks added to the source files
> I am sure you could do someway around this

I'm not sure if there's another way. I'll look at it again. Here's what I found: https://stackoverflow.com/questions/9901082/what-is-this-javascript-require

**Comment by williamdes** [comment]
Working on making tests pass and remove source files modifications

**Comment by williamdes** [comment]
It works !!!
Hours of debugging and I found how to make it work, sleep time now ^^
https://github.com/shucon/phpmyadmin/runs/1752240898?check_suite_focus=true#step:7:20

```
yarn run v1.22.5
$ jest
PASS phpMyAdmin test/javascript/sql.test.js
  ✓ test1 (2 ms)
ignored-filename /home/runner/work/phpmyadmin/phpmyadmin/test/test-env.js
loaded-module: "CommonParams", from /home/runner/work/phpmyadmin/phpmyadmin/js/src/common.js
loaded-module: "AJAX", from /home/runner/work/phpmyadmin/phpmyadmin/js/src/ajax.js
ignored-filename /home/runner/work/phpmyadmin/phpmyadmin/js/vendor/stickyfill.min.js
ignored-filename /home/runner/work/phpmyadmin/phpmyadmin/test/javascript/sql.test.js
loaded-module: "Sql", from /home/runner/work/phpmyadmin/phpmyadmin/js/src/sql.js

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.025 s
Ran all test suites.
Done in 1.78s.

```

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Nice work :100:

**Review by williamdes — APPROVED** [review]
Merging soon, after a cleanup of console.logs

**Comment by williamdes** [comment]
I made more work than expected:
- Added module aliases, no more relative paths
- Made coverage work
- Created a folder for jest specific configs
- Made the first test have some sense
- Adjusted the JS linter

**Pull request merged**